azurerm/resource_arm_key_vault_access_policy.go

```diff
@@ -0,0 +1,86 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2016-10-01/keyvault"
+	"github.com/hashicorp/terraform/helper/schema"
+	uuid "github.com/satori/go.uuid"
+)
+
+func resourceArmKeyVaultAccessPolicy() *schema.Resource {
+	// The Azure SDK doesn't provide direct access to KV access policies,
+	// so changing access policies requires a full Key Vault resource
+	// be provided. This is done here by reusing the resourceArmKeyVault
+	// schema with some tweaks so that unneed properties need not be
+	// specified in the template.
+
+	kvapSchema := resourceArmKeyVault()
```

Comment:
Doing this will make it look just like a Key Vault resource so the examples you have wouldn't follow this pater.  This resource would have to look like

```
resource "azurerm_key_vault" "test" {
  name                = "testvault"
  resource_group_name = "${azurerm_resource_group.test.name}"

  tenant_id = "d6e396d0-5584-41dc-9fc0-268df99bc610"

  access_policy {
    tenant_id = "d6e396d0-5584-41dc-9fc0-268df99bc610"
    object_id = "d746815a-0433-4a21-b95d-fc437d2d475b"

    key_permissions = [
      "get",
    ]

    secret_permissions = [
      "get",
    ]
  }
```

Which wouldn't really be allowing you to attach it to an existing keyvault.

Reply:
I missed that the SDK has the UpdateAccessPolicy function. I therefore intentionally reused the KV resource with the idea of reading all its existing properties and editing only the access policies. Given that UpdateAccessPolicy is there, I will quickly reassess approaches and write back.